src/algebra/group_ring_action.lean

```diff
@@ -65,6 +65,19 @@ def mul_semiring_action.to_ring_equiv [mul_semiring_action G R] (x : G) : R ≃+
 { .. distrib_mul_action.to_add_equiv R x,
   .. mul_semiring_action.to_ring_hom G R x }
 
+/-- A multiplicative action of `N` on `B` and a ring homomorphism `K →+* N` induce
+a multiplicative action of `K` on `B`. -/
+
+def mul_semiring_action.comp_hom (g : K →+* N) [mul_semiring_action N B] :
+mul_semiring_action K B :=
+{ smul := has_smul.comp.smul g,
+  one_smul := λ a, by simp only [has_smul.comp.smul, map_one, one_smul],
+  smul_zero:= λ n, by simp only [has_smul.comp.smul, smul_zero],
+  smul_one := λ x, by simp [has_smul.comp.smul, smul_one],
+  smul_mul := λ n x y, by simp [has_smul.comp.smul, smul_mul'],
+  smul_add := λ n x y, by simp [has_smul.comp.smul, smul_add],
+  mul_smul := λ x y a, by simp only [has_smul.comp.smul, map_mul,mul_smul (g x) (g y) a] }
```

Comment:
You should be able to just glue together the existing proofs:
```suggestion
def mul_semiring_action.comp_hom (g : K →+* N) [mul_semiring_action N B] :
mul_semiring_action K B :=
{ smul := has_smul.comp.smul g,
  ..distrib_mul_action.comp_hom g,to_monoid_hom,
  ..mul_distrib_mul_action.comp_hom g,to_monoid_hom }
```

Reply:
I 've tried doing that but this doesn't compile - I get errors about the constructor ```invalid structure value { ... }, field 'one_smul' was not provided ``` etc. Is there something else I can try?

Also, I don't know what I did but I somehow mistakenly requested a review from the maintainers. Sorry about that